sql/core/src/main/scala/org/apache/spark/sql/sources/ddl.scala

```diff
@@ -120,8 +122,10 @@ private[sql] class DDLParser extends StandardTokenParsers with PackratParsers wi
   protected lazy val pair: Parser[(String, String)] = ident ~ stringLit ^^ { case k ~ v => (k,v) }
 
   protected lazy val column: Parser[StructField] =
-    ident ~ dataType ^^ { case columnName ~ typ =>
-      StructField(columnName, typ)
+    ident ~ dataType ~ (COMMENT ~> stringLit).?  ^^ { case columnName ~ typ ~ cm =>
+      val metadata = new MetadataBuilder()
+          .putString(COMMENT.str.toLowerCase(), cm.getOrElse("")).build()
+      StructField(columnName, typ, true, metadata)
```

Comment:
How about we do not define the metadata if a comment is not provided? I feel we are losing information if we use an empty string when a comment is not provided.